core/src/main/java/hudson/model/Job.java

```diff
@@ -316,8 +318,56 @@ public boolean isBuilding() {
      */
     public boolean isLogUpdated() {
         RunT b = getLastBuild();
-        return b!=null && b.isLogUpdated();
-    }    
+        return b != null && b.isLogUpdated() || isCurrentWorkUnit();
```

Comment:
Hmmm on considering this point I think we need to add this into `isBuilding()` also. Both of the methods are supposed to return `true` if the project is building. If we are adding this to `isBuilding()` though, it may make sense to track the current work units that do not yet have a Run in a concurrent collection so that we can avoid iterating all the executors.